src/js/server/devpmt-ppt-dispatchers.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * Dispatchers for PPT pages and workflow
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.devpmt.ppt");
+
+gpii.devpmt.ppt.checkAuthorization = function (that, req, res /*, next */) {
```

Comment:
This `checkAuthorization` function is a mixture of checking authorization and performing actual actions such as redirection at certain circumstances. I feel this type of function should only perform the former and leave the latter to caller functions.

The same comment applies to other `checkAuthorization` functions.

Reply:
This is a good idea. I separated the login check into a separate function `gpii.devpmt.ppt.isLoggedIn`, and then call that in the checkAuth wrapper which still needs to perform the necessary redirects.